src/ir/decorators.py

```diff
@@ -0,0 +1,4 @@
+def two_way_subtyping(is_subtype):
```

Comment:
Move this function to the `type_utils.py` file.

Reply:
Not possible because of non-bypassable circular import.